src/plugins/profiling/server/routes/search_flamechart.ts

```diff
@@ -112,96 +123,178 @@ export function registerFlameChartSearchRoute(router: IRouter<DataRequestHandler
         // eslint-disable-next-line no-console
         console.log('EventsIndex', eventsIndex);
 
+        // Using filter_path is less readable and scrollSearch seems to be buggy - it
+        // applies filter_path only to the first array of results, but not on the following arrays.
+        // The downside os `_source` is: it takes 2.5x more time on the ES side (see "took" field).
+        // The `composite` keyword skips sorting the buckets as and return results 'as is'.
+        // A max bucket size of 100000 needs a cluster level setting "search.max_buckets: 100000".
+        const start1 = now();
         const resEvents = await esClient.search({
           index: eventsIndex.name,
-          body: {
-            size: 0,
-            query: filter,
-            aggs: {
-              group_by: {
-                terms: {
-                  field: 'StackTraceID',
-                  size: topN,
-                },
-                aggs: {
-                  sum_count: {
-                    sum: {
-                      field: 'Count',
+          size: 0,
+          query: filter,
+          aggs: {
+            group_by: {
+              composite: {
+                size: 100000, // This is the upper limit of entries per event index.
+                sources: [
+                  {
+                    traceid: {
+                      terms: {
+                        field: 'StackTraceID',
+                      },
```

Comment:
This makes sense, but are there any subtle differences that I may be missing? What's the difference between the grouping in the original versus the new one?

Reply:
The differences are
- Before, `size: topN` did not fetch all events but just the topN (=200), which is statistically very off. We need to fetch all events, as the aggregation is done by the root frame in a later step. And only after this aggregation, we know the topN graphs to be displayed. Btw, we introduced `topN` only to work-around the Kibana flamechart slowness (and memory consumption).
- Since we have to read all events in filter range, we do not need sorted buckets. The `composite` aggregation does exactly this (no sort order). This speeds up the query (by 50% in my local test).
- Because of down-sampling, we always end up with 20k..100k of events. `composite` allows to return them all by setting `size: 100000` (this needs a cluster setting of `search.max_buckets: 100000`